ChangeLog.d/fix-build-netbsd.txt

```diff
@@ -0,0 +1,4 @@
+Bugfix
+   * Fix building library/net_sockets.c and the ssl_mail_client program on
+     NetBSD. Contributed by Nia Alarie in #3422. NetBSD conditionals were added
+     for the backport because of an earlier decision not to backport the fix.
```

Comment:
The conditionals are to avoid the risk of breaking a platform that isn't NetBSD, not “because of an earlier decision”.

Also feel free to credit yourself for the contribution. E.g.

> Original fix contributed by Nia Alarie in #3422. Adapted for long-time support branches by gufe44 in #3558.